runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkPipelineRunner.java

```diff
@@ -0,0 +1,212 @@
+ * to the environment and terminates.
+ *
+ * <p>Finally Flink launches the job.
+ */
+public class LyftFlinkPipelineRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(LyftFlinkPipelineRunner.class);
+  private static String DRIVER_CMD_FLAGS = "--job_endpoint=localhost:%s";
+
+  private final String driverCmd;
+  private FlinkJobServerDriver driver = null;
+  private Thread driverThread = null;
+  private int jobPort = 55555;
+
+  public LyftFlinkPipelineRunner(String driverCmd) {
+    this.driverCmd = driverCmd;
+  }
+
+  /** Main method to be called within the Flink OptimizerPlanEnvironment. */
+  public static void main(String[] args) throws Exception {
+    Preconditions.checkArgument(
+        ExecutionEnvironment.getExecutionEnvironment() instanceof OptimizerPlanEnvironment,
+        "Can only execute in OptimizerPlanEnvironment");
+    LOG.info("entry points args: %s", Arrays.asList(args));
+    EntryPointConfiguration configuration = parseArgs(args);
+    LyftFlinkPipelineRunner runner = new LyftFlinkPipelineRunner(configuration.driverCmd);
+    try {
+      runner.startJobService();
+      runner.runDriverProgram();
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("Job %s failed.", configuration.driverCmd), e);
+    } finally {
+      LOG.info("Stopping job service");
+      runner.stopJobService();
+    }
+    LOG.info("Job submitted successfully.");
+  }
+
+  private static class EntryPointConfiguration {
+    @Option(
+        name = "--driver-cmd",
+        required = true,
+        usage =
+            "Command that launches the Python driver program. "
+                + "(The job service endpoint will be appended as --job_endpoint=localhost:<port>.)")
+    private String driverCmd = null;
+  }
+
+  private static EntryPointConfiguration parseArgs(String[] args) {
+    EntryPointConfiguration configuration = new EntryPointConfiguration();
+    CmdLineParser parser = new CmdLineParser(configuration);
+    try {
+      parser.parseArgument(args);
+    } catch (CmdLineException e) {
+      LOG.error("Unable to parse command line arguments.", e);
+      parser.printUsage(System.err);
+      throw new IllegalArgumentException("Unable to parse command line arguments.", e);
+    }
+    return configuration;
+  }
+
+  private void startJobService() throws Exception {
+    final PrintStream oldErr = System.err;
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream newErr = new PrintStream(baos);
+    try {
+      System.setErr(newErr);
+      driver =
+          FlinkJobServerDriver.fromParams(
+              new String[] {"--job-port=" + jobPort, "--artifact-port=0", "--expansion-port=0"});
+      driverThread = new Thread(driver);
+      driverThread.start();
+      boolean success = false;
+
+      // TODO: check for job service ready
+      Thread.sleep(5000);
+      success = true;
```

Comment:
This needs to be addressed.

Reply:
I think we could use something like the option retrieval to check for the lifeness of the JobService.

Reply:
Otherwise I would remove these two lines here and just process the stdout below.